scompose/project/project.py

```diff
@@ -411,50 +434,38 @@ def _create(
             # Generate shared hosts file
             hosts_file = self.create_hosts(lookup)
 
-        # First create those with no dependencies
-        while names:
-
-            for instance in self.iter_instances(names):
-
-                # Flag to indicated create
-                do_create = True
-
-                # Ensure created, skip over if not
-                depends_on = instance.params.get("depends_on", [])
-                for depends_on in depends_on:
-                    if depends_on not in created:
-                        count += 1
-                        do_create = False
-
-                if do_create:
-
-                    # Generate a resolv.conf to bind to the container
-                    if self.sudo and not no_resolv:
-                        resolv = self.generate_resolv_conf()
-                        instance.volumes.append("%s:/etc/resolv.conf" % resolv)
-
-                        # Create a hosts file for the instance based, add as volume
-                        instance.volumes.append("%s:/etc/hosts" % hosts_file)
+        for instance in self.iter_instances(names):
+            depends_on = instance.params.get("depends_on", [])
+            for dep in depends_on:
```

Comment:
I don't understand this logic here - if a dependency isn't created yet, how can you be sure that it's circular?

Reply:
Because the `instances` list is sorted out in such a way that if the dependencies are not a DAG, at least one dependency will be missing.

Let's take this singularity-compose file as an example:
```
version: "1.0"
instances:
  client0:
    build:
      context: ./images
    depends_on: [client2]
  client1:
    build:
      context: ./images
    depends_on: [client2]
  client2:
    build:
      context: ./images
```
The instances list will be sorted out in this way `[client2, client0, client1]`, so that `client2` will be created before the dependent containers.


Let's consider a wrong singularity-compose file instead:
```
version: "1.0"
instances:
  client0:
    build:
      context: ./images
    depends_on: [client2]
  client1:
    build:
      context: ./images
    depends_on: [client2]
  client2:
    build:
      context: ./images
      depends_on: [client0]
```
The instances list will be sorted out in this way `[client0, client1, client2]`, so that `client0` will not run is `client2` is not up already: that will raise the circular dependency error.

Reply:
The sorting happens [on this line](https://github.com/singularityhub/singularity-compose/blob/894e4eb2ff2faed44ba027bacde676d0c065cb6b/scompose/project/project.py#L171)

Reply:
ah okay - do we have an easy way to add a simple test with a file like this?